chore: Introduces TransactionOptionsInterface

JirkaChadima opened the pull request:
Improves type-checking and fixes some inconsistencies.

MatiasOS commented on src/data-model/on-chain-hotel.js:
```diff
@@ -252,11 +252,11 @@ class OnChainHotel implements HotelInterface {
    *
    * @param {WalletInterface} wallet that signs the transaction
-   * @param {Object} transactionOptions usually contains from and to. Gas is automatically computed.
+   * @param {TransactionOptionsInterface} transactionOptions usually contains from and to. Gas is automatically computed.
```
What do you think about something like "must have `from`, and optionally `to`, `gas` and `value`"

JirkaChadima replied:
They are optional. That's defined i the interface with gas?  Yhe question mark means optional

JirkaChadima commented:
@MatiasOS I've changed the docstrings and realized that we're actually not using `to` property at all, so I removed that as well. Is the meaning more clear now?

JirkaChadima commented:
Ping @MatiasOS

MatiasOS commented:
Yes, I like it now. I got confused by the difference between the definition of the interface and the comment. It seems much clearer to me!
